ministryofjustice/laa-apply-for-criminal-legal-aid
# CRIMAPP-542 Court of Appeal pages

## Description of change

Update the court of appeals pages:
* `appeal_details` page - the appeal changes details question and the MAAT ID question were removed and replaced with 'Has an original application been submitted?' question. If they say 'yes' they are redirected to the below page, if 'no' then the application goes through the date stamp logic
* Added `appeal_financial_circumstances` page, if the user selects 'Yes' then `appeal_with_changes_details' text area appears. If 'Yes' is selected, the next page is date stamp/enter address. If 'No' then they proceed to the below page
* Added `appeal_reference_number` page where the user enters either the MAAT ID or USN
* Updated the final Review page 

## Link to relevant ticket

[CRIMAPP-542](https://dsdmoj.atlassian.net/browse/CRIMAPP-542)

## Notes for reviewer

## Screenshots of changes (if applicable)

### Before changes:

<img width="332" alt="Screenshot 2024-04-11 at 19 05 24" src="https://github.com/ministryofjustice/laa-apply-for-criminal-legal-aid/assets/15728561/b65c521a-5aa8-446f-8f77-dfc746cef6ff">


<img width="332" alt="Screenshot 2024-04-11 at 19 05 52" src="https://github.com/ministryofjustice/laa-apply-for-criminal-legal-aid/assets/15728561/8bc9777e-53d9-412a-995b-3194b5695262">

<img width="332" alt="Screenshot 2024-04-11 at 19 05 40" src="https://github.com/ministryofjustice/laa-apply-for-criminal-legal-aid/assets/15728561/9c59dd2a-8cec-44ef-841f-494243ef3cc0">

### After changes:
<img width="322" alt="Screenshot 2024-04-11 at 18 53 11" src="https://github.com/ministryofjustice/laa-apply-for-criminal-legal-aid/assets/15728561/5ba5c5bd-f9da-4469-95fc-de4902351d09">

<img width="376" alt="Screenshot 2024-04-11 at 18 51 06" src="https://github.com/ministryofjustice/laa-apply-for-criminal-legal-aid/assets/15728561/003cd17c-dcde-48ba-9b82-1f35f270d581">

## How to manually test the feature

Start a new application, when you get to the 'case_type' page, select 'Appeal to crown court'. Continue through the following screens to the Review page. Go back and change your answers and check that the correct fields still appear. 

[CRIMAPP-542]: https://dsdmoj.atlassian.net/browse/CRIMAPP-542?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
